src/test/java/net/minecraftforge/debug/misc/DataPackRegistriesTest.java

```diff
@@ -0,0 +1,228 @@
+ * Data is loaded from the following folders in test resources:</p>
+ * <ul>
+ * <li><code>data/data_pack_registries_test/data_pack_registries_test/unsyncable/test.json</code></li>
+ * <li><code>data/data_pack_registries_test/tags/data_pack_registries_test/unsyncable/test.json</code></li>
+ * <li><code>data/data_pack_registries_test/data_pack_registries_test/syncable/test.json</code></li>
+ * <li><code>data/data_pack_registries_test/tags/data_pack_registries_test/syncable/test.json</code></li>
+ * </ul>
+ */
+@Mod(DataPackRegistriesTest.MODID)
+public class DataPackRegistriesTest
+{
+    private static final boolean ENABLED = true;
+    public static final String MODID = "data_pack_registries_test";
+    private static final Logger LOGGER = LogUtils.getLogger();
+    private static final ResourceLocation TEST_RL = new ResourceLocation(MODID, "test");
+    
+    private final RegistryObject<Unsyncable> datagenTestObject;
+
+    public DataPackRegistriesTest()
+    {
+        if (ENABLED)
+        {
+            final IEventBus modBus = FMLJavaModLoadingContext.get().getModEventBus();
+            final IEventBus forgeBus = MinecraftForge.EVENT_BUS;
+            
+            // Deferred Registers can be created for datapack registries in static init or mod constructor.
+            // (We'll do it in mod constructor, as when doing the ENABLED check it's less verbose than static init.)
+            // As with static registries, any mod can make a Deferred Register for a given datapack registry,
+            // but only one mod can register the internal registry with makeRegistry.
+            final DeferredRegister<Unsyncable> unsyncables = DeferredRegister.create(Unsyncable.REGISTRY_KEY, MODID);
+            final DeferredRegister<Syncable> syncables = DeferredRegister.create(Syncable.REGISTRY_KEY, MODID);
+            
+            // RegistryBuilder#dataPackRegistry marks the registry as a datapack registry rather than a static registry.
+            unsyncables.makeRegistry(Unsyncable.class,
+                () -> new RegistryBuilder<Unsyncable>().disableSaving().dataPackRegistry(Unsyncable.DIRECT_CODEC));
+            // The overload of #dataPackRegistry that takes a second codec marks the datapack registry as syncable.
+            syncables.makeRegistry(Syncable.class,
+                () -> new RegistryBuilder<Syncable>().disableSaving().dataPackRegistry(Syncable.DIRECT_CODEC, Syncable.DIRECT_CODEC));
+            
+            // Datapack registry elements can be datagenerated, but they must be registered as builtin objects first.
+            this.datagenTestObject = unsyncables.register("datagen_test", () -> new Unsyncable("Datagen Success"));
+            
+            unsyncables.register(modBus);
+            syncables.register(modBus);
+            
+            modBus.addListener(this::onGatherData);
+            forgeBus.addListener(this::onServerStarting);            
```

Comment:
Formatting: there should be no extraneous trailing whitespace.
```suggestion
            forgeBus.addListener(this::onServerStarting);
```

Reply:
Fixed in next iteration